internal/yamlutil/decoder.go

```diff
@@ -0,0 +1,351 @@
+/*
```

Comment:
This is more of a nit than anything else, but could you move this code to a designated 3rd party namespace to make it easier to see that this is upstream code?

Reply:
Fixed. I think this was more than a nit.